packages/airnode-validator/src/convertor.ts

```diff
@@ -1,66 +1,75 @@
-import fs from 'fs';
-
-import { Result } from './types';
+import { Log, Result } from './types';
 import * as logger from './utils/logger';
 import { processSpecs } from './processor';
-
-const path = require('path');
+import * as utils from './commands/utils';
 
 /**
  * Converts a specification according to the template
  * @param specsPath - specification file to convert, root must be an object (not an array)
  * @param templatePath - template json file
+ * @param interpolatePath - path to env file that will be interpolated with specification file
  * @returns array of messages and converted specification
  */
-export function convert(specsPath: string | undefined, templatePath: string | undefined): Result {
+export function convert(
+  specsPath: string | undefined,
+  templatePath: string | undefined,
+  interpolatePath?: string
+): Result {
   if (!specsPath || !templatePath) {
     return { valid: false, messages: [logger.error('Specification and template file must be provided')], output: {} };
   }
 
   let template, specs;
+  const messages: Log[] = [];
 
-  try {
-    template = fs.readFileSync(path.resolve(templatePath), 'utf-8');
-  } catch (e) {
-    return { valid: false, messages: [logger.error(`Unable to read file ${path.resolve(templatePath)}`)], output: {} };
+  if ((template = utils.readJson(templatePath, messages)) === undefined) {
```

Comment:
Also, not a big fan of assignments within an `if` statement. I find it unnecessarily confusing at first glance.

Reply:
I think this comes down to personal preference. If it would be used in more complex condition I would agree with you, but I believe this condition is simple enough and it is easy to understand, moving it out of the condition would make it unnecessarily long, so I would prefer leaving it as is.